staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -241,15 +242,16 @@ func (c *threadSafeMap) AddIndexers(newIndexers Indexers) error {
 
 // updateIndices modifies the objects location in the managed indexes, if this is an update, you must provide an oldObj
 // updateIndices must be called from a function that already has a lock on the cache
-func (c *threadSafeMap) updateIndices(oldObj interface{}, newObj interface{}, key string) error {
+func (c *threadSafeMap) updateIndices(oldObj interface{}, newObj interface{}, key string) {
```

Comment:
surprising that we did not use this returned error anywhere.

Reply:
> surprising that we did not use this returned error anywhere.

Likewise.  Given what we use the indexers for, it may be more appropriate to panic in the callers than force general, "I'm going to try continuing" behavior using `HandleError` below.

Reply:
@deads2k what do you ultimately want to do here? I hate panicking from a library... maybe we should actually make the callers look at the returned error?

Reply:
If getting a error from the function results in a panic then maybe we should change the signature of the function? Remove the `error` return value? That is a breaking change of course, same as panicking on error though.

Reply:
> ncdc 2 days ago Member
> @deads2k what do you ultimately want to do here? I hate panicking from a library... maybe we should actually make the callers look at the returned error?

We could make callers handle it, but I'd guess that the index is hosed/untrustworthy if this happens.  They may choose to kill themselves.

Reply:
Meh. If we make the callers deal with errors, that means modifying the Add/Update/Delete/Replace signatures in the interface to return errors. Maybe it's just best to panic after all.

Reply:
@ncdc ok, I made it panic. I think anything is better than status quo.